### DISCUSSION TITLE
Check is code-server is working

### DISCUSSION THREAD

**Discussion by paskalnikita** [coder/code-server#5582] [Q&A]
Hello, I did `wget tar.gz ` code-server and did entrypoint `bin/code-server` on my `docker image`. I did start container and got this logs:

`code-server    | [2022-09-23T15:13:56.872Z] info  code-server 4.7.0 7f108c61ab670ccaae4cfde5de2909c003db1ce8
code-server    | [2022-09-23T15:13:56.873Z] info  Using user-data-dir ~/.local/share/code-server
code-server    | [2022-09-23T15:13:56.882Z] info  Using config file ~/.config/code-server/config.yaml
code-server    | [2022-09-23T15:13:56.883Z] info  HTTP server listening on http://127.0.0.1:8080/
code-server    | [2022-09-23T15:13:56.883Z] info    - Authentication is enabled
code-server    | [2022-09-23T15:13:56.883Z] info      - Using password from $PASSWORD
code-server    | [2022-09-23T15:13:56.883Z] info    - Not serving HTTPS`

But whenever I am trying to enter  http://127.0.0.1:8080/ i am getting blank page without any logs appearing on console, should I run install or whatever to being able to use it?

**Comment by jsjoeio** [comment]
Hmm seems like it should work. Silly question, is port 8080 exposed for you to access outside of the container? See docs [here](https://coder.com/docs/code-server/latest/install#docker).

**Reply by lg-nineoneninethreenine** [reply]
@jsjoeio doesn't Docker have its own IP Address? 
Suggestion: maybe type `ip addr` in the host system and grep for docker?

**Reply by paskalnikita** [reply]
> is port 8080 exposed

yes, I am doing docker-compose up --build every time. I have ` version: "2.1" `and 
`ports:
      - 8443:8443
      - 8080:8080
   `

**Reply by paskalnikita** [reply]
> @jsjoeio doesn't Docker have its own IP Address? Suggestion: maybe type `ip addr` in the host system and grep for docker?

`# ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: tunl0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/ipip 0.0.0.0 brd 0.0.0.0
3: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/sit 0.0.0.0 brd 0.0.0.0
584: eth0@if585: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
    link/ether 02:42:ac:15:00:02 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 172.21.0.2/16 brd 172.21.255.255 scope global eth0
       valid_lft forever preferred_lft forever`

**Reply by lg-nineoneninethreenine** [reply]
HA! Got it on my Pop!_ OS PC. The IP is `172.17.0.1`
Console:
```
❯ ip addr | grep docker
5: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
    inet 172.17.0.1/16 brd 172.17.255.255 scope global docker0
```
Ran on zsh btw so all the fancy things

**Reply by lg-nineoneninethreenine** [reply]
That's what you could mostly need. If there is anything else, try exposing the port like @jsjoeio said, if it isn't exposed.

Not to the internet btw, to your local PC/WiFi, I suspect.

**Reply by lg-nineoneninethreenine** [reply]
Hold on, let's backtrack for a moment...

```
# ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000 link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00 inet 127.0.0.1/8 scope host lo valid_lft forever preferred_lft forever 2: tunl0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000 link/ipip 0.0.0.0 brd 0.0.0.0 3: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000 link/sit 0.0.0.0 brd 0.0.0.0 584: eth0@if585: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default link/ether 02:42:ac:15:00:02 brd ff:ff:ff:ff:ff:ff link-netnsid 0 inet 172.21.0.2/16 brd 172.21.255.255 scope global eth0 valid_lft forever preferred_lft forever
```
There should be a fifth scope like my computer. Are you positive you have the latest Docker version(Docker version 20.10.18, build b40c2f6)?

**Reply by paskalnikita** [reply]
> Hold on, let's backtrack for a moment...
> 
> ```
> # ip addr
> 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000 link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00 inet 127.0.0.1/8 scope host lo valid_lft forever preferred_lft forever 2: tunl0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000 link/ipip 0.0.0.0 brd 0.0.0.0 3: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000 link/sit 0.0.0.0 brd 0.0.0.0 584: eth0@if585: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default link/ether 02:42:ac:15:00:02 brd ff:ff:ff:ff:ff:ff link-netnsid 0 inet 172.21.0.2/16 brd 172.21.255.255 scope global eth0 valid_lft forever preferred_lft forever
> ```
> 
> There should be a fifth scope like my computer. Are you positive you have the latest Docker version(Docker version 20.10.18, build b40c2f6)?
Btw entering 172.17.0.1 gives me the same - blank page. I am on ` Docker version 20.10.16, build aa7e414`

**Reply by paskalnikita** [reply]
@im-coder-lg I did update Docker to latest. Nothing have changed
My Dockerfile:

`
FROM ubuntu:20.04
RUN apt-get update && apt-get install wget -y && apt-get install git -y && apt-get install curl -y

RUN wget https://github.com/codercom/code-server/releases/download/v4.7.0/code-server-4.7.0-linux-amd64.tar.gz \
    && tar -xzvf code-server-4.7.0-linux-amd64.tar.gz && chmod +x code-server-4.7.0-linux-amd64/code-server
RUN mkdir -p /.config/code-server
COPY config.yaml /.config/code-server/config.yaml
ADD ./code /code
EXPOSE 8080
EXPOSE 8443

ENTRYPOINT ["code-server-4.7.0-linux-amd64/code-server"]
`

I do launch container with a help of docker-compose up --build

`
version: "2.1"
services:
  code-server:
    build: .
    container_name: code-server
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/London
      - PASSWORD=password
    ports:
      - 8443:8443
      - 8080:8080
    restart: unless-stopped
`

[jsjoeio](https://github.com/jsjoeio) I have read documentation and after

I did ` docker build ` later have run container with this command:
` docker run -it --name code-server -p 127.0.0.1:8080:8080 code-server:latest ` and it did the same, blank page.  
I am using windows btw...

**Reply by paskalnikita** [reply]
I also did execute all that stuff on wsl(ubuntu). and curl 127.0.0.1:8080 gave me curl: (52) Empty reply from server 
Status from the container told that server is running on .... but got the same result

**Reply by paskalnikita** [reply]
So, it seems to be maybe wrong configuration for localhost or smt. like that, because when I do use https://hub.docker.com/r/linuxserver/code-server with docker-compose everything works fine... :(

**Reply by jsjoeio** [reply]
> So, it seems to be maybe wrong configuration for localhost or smt. like that, because when I do use https://hub.docker.com/r/linuxserver/code-server with docker-compose everything works fine... 

Hmm 🤔 That's good to know. @code-asher sorry to pull you into a long thread, any ideas?

**Reply by code-asher** [reply]
Make sure to bind to 0.0.0.0 otherwise code-server cannot be seen outside the Docker container.  For example `code-server --bind-addr 0.0.0.0:8080`.
